### DISCUSSION TITLE
Command to run current test?

### DISCUSSION THREAD

**Discussion by joakim-hove** [mrcjkb/rustaceanvim#497] [Q&A] [answered]
I have just started using rustaceanvim, mainly for the integration with testing.

Currently I use`:RustLsp testable` to get a list of tests and then select a test for running.

Ideally I would like a function to run the the "current" test - as in which test the cursor is currently in? Is such a function available? As an added bonus similar functions for all tests in file/module and all tests in crate would be nice?

**Comment by mrcjkb** [comment, marked as answer]
hey 👋 

You can do this by enabling code lenses (`:h lsp-codelens`) or by using the [neotest](https://github.com/nvim-neotest/neotest) adapter.

> As an added bonus similar functions for all tests in file/module and all tests in crate would be nice?

Something like that would need to be implemented in rust-analyzer, not rustaceanvim.

**Reply by joakim-hove** [reply]
Thank you - I will look into your suggestions.

**Reply by joakim-hove** [reply]
I understand that this is outside the rustaceanvim plugin, but anyway I struggle with both of the alternatives. The `:h lsp-codelens` is unfortunately still a bit above my paygrade when it comes to nevim/lua - but I can not seem to find any command to "just turn on code lenses". Any hint for that?

**Reply by mrcjkb** [reply]
https://neovim.io/doc/user/lsp.html#vim.lsp.codelens.run()
